File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int max_length=0;
        int max_freq=0;
        int l=0;
        unordered_map<char,int> mp;
        for(int r=0;r<s.size();r++)
        {
            char curr_string=s[r];
            mp[curr_string]++;
            max_freq=max(max_freq,mp[curr_string]);
            int length=r-l+1;
            if((length-max_freq)>k)
            {
                char left_str=s[l];
                mp[left_str]--;
                l++;
                
            }
            max_length=max(max_length,r-l+1);

        }
        return max_length;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        st=set(nums)
        longest=0
        for x in st:
            if x-1 not in st:
                length=1
                while x+length in st:
                    length+=1
                longest=max(longest,length)
        return longest






        


        